### PR TITLE
Clarified usage of application domains in NSB host

### PR DESCRIPTION
**Pull request by SzymonPobiega**

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
@SimonCropp @johnsimons can you review?

**Comment by johnsimons** [comment]
:+1:

File: nservicebus/hosting/nservicebus-host/index.md [hunk 1]
```diff
@@ -34,6 +34,11 @@ If you want to avoid the scanning process you can explicitly configure the type
 
 <!-- import ExplicitHostConfigType -->
 
+## Application Domains
+
+The `NServiceBus.Host.exe` creates a separate *service* [Application Domain](https://msdn.microsoft.com/en-us/library/2bh4z9hs(v=vs.110).aspx) to run NServiceBus and the user code. The new domain is assigned a configuration file named after the dll that contains the class implementing `IConfigureThisEndpoint`. All the configuration should be done in that file (as opposed to `NServiceBus.Host.exe.config`). In most cases that means just adding the `app.config` file to the project and letting MSBuild take care of renaming it while moving to the `bin` folder.
```
Comment (SimonCropp) [inline nservicebus/hosting/nservicebus-host/index.md]: remove the `(v=vs.110)` we dont want to be version specific

File: nservicebus/hosting/nservicebus-host/index.md [hunk 2]
```diff
@@ -34,6 +34,11 @@ If you want to avoid the scanning process you can explicitly configure the type
...
+
+NOTE: When the endpoint configuration is not specified explicitly, the host scans all the assemblies to find it and it does so in the context of the *host* application domain, not the new *service* domain. Because of that, when [reditecting assembly versions](https://msdn.microsoft.com/en-us/library/7wd6ex19(v=vs.110).aspx), the `assemblyBinding` element needs to be present in both `NServiceBus.Host.exe.config` and `app.config`.
```
Comment (SimonCropp) [inline nservicebus/hosting/nservicebus-host/index.md]: remove `(v=vs.110)`

**Pull request merged**